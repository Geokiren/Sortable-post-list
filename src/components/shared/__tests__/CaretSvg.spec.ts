import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import CaretSvg from '../CaretSvg.vue';

describe('CaretSvg', () => {
  const wrapper = mount(CaretSvg, { props: { classes: 'fill-red' } });

  it('Renders properly with classes through props', () => {
    expect(wrapper.classes()).toContain('fill-red');
  });

  it('Renders properly with the correct element tag', () => {
    const tag = wrapper.element.tagName.toLowerCase();

    expect(tag).toBe('svg');
  });

  it('Matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
