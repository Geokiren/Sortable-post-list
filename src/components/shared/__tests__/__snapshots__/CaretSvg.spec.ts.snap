// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CaretSvg > Matches snapshot 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="fill-red" data-test="caret-svg">
  <path d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"></path>
</svg>"
`;
